src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -429,6 +477,17 @@ private JsonRpcError ReadError(JToken json)
             };
         }
 
+        private void ClearProgressObject(long id)
+        {
+            long progressId = 0;
+
+            if (this.requestProgressMap.TryGetValue(id, out progressId))
+            {
+                this.requestProgressMap.Remove(id);
+                this.progressMap.Remove(progressId);
+            }
+        }
```

Comment:
You're probably going to need a lock around this, since while you're receiving one message you could be sending a message on another thread, which accesses these same data structures.